### PR TITLE
Replace Legacy Request Data in Pricing Admin

### PR DESCRIPTION
**Pull request by tn3rb**
Similar to previous PRs, this one:

- replaces all uses of the legacy admin request data with the new Request class
- autoformats and cleans up the code

### CODE REVIEW

File: caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php [hunk 1]
```diff
@@ -748,280 +723,233 @@ protected function _insert_or_update_price($insert = false)
...
+
+        $all_updated = true;
+        foreach ($row_ids as $i => $row_id) {
+            // Update the prices when re-ordering
+            $fields_n_values = ['PRC_order' => $i + 1];
+            $query_params    = [['PRC_ID' => absint($row_id)]];
+            // any failure will toggle $all_updated to false
+            $all_updated =
+                $row_id && EEM_Price::instance()->update($fields_n_values, $query_params) !== false ? $all_updated
+                    : false;
```
Comment (knazart) [inline caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php:855]:
```suggestion
            $all_updated = $row_id && EEM_Price::instance()->update($fields_n_values, $query_params) !== false
                ? $all_updated
                : false;
```

**Review by knazart — APPROVED** [review]
Looks good to me.

File: caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php (hunk 1 above)
Comment (tn3rb) [reply caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php:855]: oh ya... missed that.. thnx 👍🏻 will fix

**Comment by tn3rb** [comment]
thnx @knazart 

@Pebblo did you get a chance to try this out with that issue you ran into?

**Comment by Pebblo** [comment]
Hey Brent,

I'm getting a fatal whenever I try to delete a price/price type on this branch:

```
[25-Mar-2022 10:27:45 UTC] PHP Fatal error:  Uncaught Error: Call to undefined method Pricing_Admin_Page::trashRestoreDeleteEntities() in C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\caffeinated\admin\new\pricing\Pricing_Admin_Page.core.php:742
Stack trace:
#0 C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\core\admin\EE_Admin_Page.core.php(1023): Pricing_Admin_Page->_trash_or_restore_price(true)
#1 C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\core\admin\EE_Admin_Page.core.php(783): EE_Admin_Page->_route_admin_request()
#2 C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\core\admin\EE_Admin_Page.core.php(606): EE_Admin_Page->route_admin_request()
#3 C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\core\admin\EE_Admin_Page.core.php(230): EE_Admin_Page->_page_setup()
#4 [internal function]: EE_Admin_Page->__construct(true)
#5 C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\core\EE_Registry.core.php(1275): ReflectionClass->newInstanceArgs(Array)
#6 C:\laragon\www\ee4\wp in C:\laragon\www\ee4\wp-content\plugins\event-espresso-core-reg\caffeinated\admin\new\pricing\Pricing_Admin_Page.core.php on line 742
```

**Comment by tn3rb** [comment]
OMGOSH! 
sry @Pebblo  I must not have pushed the changes in one of the files I edited!
it _should_ work now 😬

File: caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php [hunk 2]
```diff
@@ -1,318 +1,311 @@
...
     protected function _set_page_routes()
     {
-        $prc_id = ! empty($this->_req_data['PRC_ID']) && ! is_array($this->_req_data['PRC_ID'])
-            ? $this->_req_data['PRC_ID'] : 0;
-        $prt_id = ! empty($this->_req_data['PRT_ID']) && ! is_array($this->_req_data['PRT_ID'])
-            ? $this->_req_data['PRT_ID'] : 0;
-        $this->_page_routes = array(
-            'default'                     => array(
+        $PRC_ID             = $this->request->getRequestParam('PRC_ID', 0, DataType::INTEGER);
+        $PRT_ID             = $this->request->getRequestParam('PRT_ID', 0, DataType::INTEGER);
```
Comment (hoseinrafiei) [inline caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php:54]: Nice extraction!

File: caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php [hunk 3]
```diff
@@ -656,85 +634,82 @@ public function _edit_price_details_meta_box()
...
         // is this a new Price ?
         if ($insert) {
             // run the insert
-            if ($PRC_ID = $PRC->insert($set_column_values)) {
+            $PRC_ID = EEM_Price::instance()->insert($set_column_values);
+            if ($PRC_ID) {
                 // make sure this new price modifier is attached to the ticket but ONLY if it is not a tax type
-                $PR = EEM_price::instance()->get_one_by_ID($PRC_ID);
-                if ($PR->type_obj()->base_type() !== EEM_Price_Type::base_type_tax) {
+                $price = EEM_price::instance()->get_one_by_ID($PRC_ID);
```
Comment (hoseinrafiei) [inline caffeinated/admin/new/pricing/Pricing_Admin_Page.core.php:684]: Better naming 👍

**Review by hoseinrafiei — APPROVED** [review]
Looks perfect

**Pull request merged**